modules/core/src/matmul.simd.hpp

```diff
@@ -2493,7 +2493,27 @@ double dotProd_16s(const short* src1, const short* src2, int len)
 
 double dotProd_32s(const int* src1, const int* src2, int len)
 {
+#if CV_VSX
+    double r = 0.0;
+    int i = 0;
+    v_float64 a(0.0, 0.0);
+    v_float64 b(0.0, 0.0);
+    int vIter = len / v_int32::nlanes;
+
+    for( i = 0; i < vIter; i++ )
```

Comment:
Personally I think this should be

```
int vIter = len & -v_int32::nlanes;
for( i = 0; i < vIter; i += v_int32::nlanes )
```

And then you can remove the

```
i *= v_int32::nlanes;
```

Reply:
OpenCV vectorized loops usually looks like this:
```
for (; i <= len - v_float32::nlanes; i += v_float32::nlanes)
```